ingestion/src/metadata/ingestion/source/database/doris/metadata.py

```diff
@@ -0,0 +1,314 @@
+            children.append(
+                _get_column(k, name_type[0], name_type[1], "YES", None, None)
+            )
+    return {
+        "name": field,
+        "default": default,
+        "nullable": True if null == "YES" else None,
+        "type": _type,
+        "data_type": data_type,
+        "display_type": _type.lower(),
+        "system_data_type": system_data_type,
+        "comment": comment,
+        "ordinalPosition": ordinal,
+        "arr_data_type": arr_data_type,
+        "children": children,
+    }
+
+
+class DorisSource(CommonDbSourceService):
+    """
+    Implements the necessary methods to extract
+    Database metadata from Mysql Source
+    """
+
+    @classmethod
+    def create(cls, config_dict, metadata: OpenMetadata):
+        config: WorkflowSource = WorkflowSource.parse_obj(config_dict)
+        if config.serviceConnection is None:
+            raise InvalidSourceException("Missing service connection")
+        connection = cast(DorisConnection, config.serviceConnection.__root__.config)
+        if not isinstance(connection, DorisConnection):
+            raise InvalidSourceException(
+                f"Expected DorisConnection, but got {connection}"
+            )
+        return cls(config, metadata)
+
+    def query_table_names_and_types(
+        self, schema_name: str
+    ) -> Iterable[TableNameAndType]:
+        """
+        Connect to the source database to get the table
+        name and type. By default, use the inspector method
+        to get the names and pass the Regular type.
+
+        This is useful for sources where we need fine-grained
+        logic on how to handle table types, e.g., external, foreign,...
+        """
+        tables = [
+            TableNameAndType(name=name, type_=RELKIND_MAP.get(engine))
+            for name, engine in self.connection.execute(
+                sql.text(DORIS_GET_TABLE_NAMES), {"schema": schema_name}
+            )
+            or []
+        ]
+        return tables
+
+    @staticmethod
+    def get_table_description(
+        schema_name: str, table_name: str, inspector: Inspector
+    ) -> str:
+        description = None
+        try:
+            table_info: dict = inspector.get_table_comment(table_name, schema_name)
+        # Catch any exception without breaking the ingestion
+        except Exception as exc:  # pylint: disable=broad-except
+            logger.debug(traceback.format_exc())
+            logger.warning(
+                f"Table description error for table [{schema_name}.{table_name}]: {exc}"
+            )
+        else:
+            description = table_info.get("text")
+
+        return description[0]
+
+    def _get_columns(self, table_name, schema=None):
+        """
+        Overriding the dialect method to add raw_data_type in response
+        """
+
+        table_columns = []
+        primary_columns = []
+        # row schema: Field, Type, Collation, Null, Key, Default, Extra, Privileges, Comment
```

Comment:
Can we add a pedantic model to have a more explicit schema validation and access for the row?

Reply:
Sorry,I don't know how to do that. Could you please give me a example @TeddyCr